File: symfony-api/src/Books/Application/Insert/BookCreator.php
<?php


namespace App\Books\Application\Insert;


use App\Books\Domain\Book;
use App\Books\Domain\BookRepository;

class BookCreator
{

    private BookRepository $repository;

    public function __construct(BookRepository $repository)
    {
        $this->repository = $repository;
    }

    public function __invoke(int $id, string $title, string $description): void
    {
        $book= Book::create($id,$title,$description);
        $this->repository->save($book);
    }

}

File: gin/src/Book/Application/BookCreator.go
package Application

import "gin-ddd/src/Book/Domain"

type BookCreator struct {
	repository Domain.BookRepository
}

func NewBookCreator(repository Domain.BookRepository) *BookCreator {
	return &BookCreator{repository: repository}
}
func (bc BookCreator) Run(title string, description string) error {

	var book = Domain.CreateBook(title, description)
	err := bc.repository.Insert(book)
	return err
}
